packages/graphiql-toolkit/src/create-fetcher/lib.ts

```diff
@@ -57,14 +74,18 @@ export const isSubscriptionWithName = (
 export const createSimpleFetcher =
   (options: CreateFetcherOptions, httpFetch: typeof fetch): Fetcher =>
   async (graphQLParams: FetcherParams, fetcherOpts?: FetcherOpts) => {
+    const headers = [
+      new Headers({
+        'content-type': 'application/json',
+      }),
+      new Headers(options.headers ?? {}),
+      new Headers(fetcherOpts?.headers ?? {}),
+    ].reduce(mergeHeadersWithSetStrategy, new Headers());
```

Comment:
```suggestion
const headers = new Headers({
  'content-type': 'application/json',
  ...options.headers,
  ...fetcherOpts?.headers,
})
```
produce the same result? you can pass multiple key-values
<img width="483" alt="image" src="https://github.com/user-attachments/assets/0b94d924-e1aa-45b0-a04f-6670f86a321d" />

Reply:
The type of `fetcherOpts?.headers` is `undefined | HeadersInit`, do you agree? If so, can you see how spreading fails when the given value for `HeadersInit` is case of `Headers` for example:


```ts
> new Headers({ ...new Headers([['set-cookie','one'],['set-cookie','two']]) })
Headers {}
```

Am I missing something?

Reply:
sorry I missed the type of `options.headers` and `fetcherOpts.headers` was changed

Reply:
All good, should we consider this thread resolved?

Reply:
is following can be used instead?

```js
  const headers = new Headers(
    Object.fromEntries([
      [['content-type'], ['application/json']],
      ...(options.headers || []),
      ...(fetcherOpts.headers || [])
    ])
  )
```

Reply:
I have applied the refactor. Thanks @dimaMachina.